### PR TITLE
add requireResidentKey param to the invocation step of authenticatorMakeCredential

### PR DESCRIPTION
**Pull request by AngeloKai**
fixes https://github.com/w3c/webauthn/issues/485

***
[Preview](https://s3.amazonaws.com/pr-preview/AngeloKai/webauthn/angelo-addParam.html) | [Diff](https://s3.amazonaws.com/pr-preview/w3c/webauthn/119dd51...AngeloKai:369cab2.html)

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -28,7 +28,7 @@ Text Macro: RP Relying Party
 Text Macro: RPS Relying Parties
 Text Macro: INFORMATIVE <em>This section is not normative.</em>
 Text Macro: WAC WebAuthn Client
-Ignored Vars: op, alg, type, algorithm
+Ignored Vars: op, alg, type, algorithm, authenticatorSelection
```
Comment (equalsJeffH) [inline index.bs]: why add `authenticatorSelection` here?

File: index.bs [hunk 2]
```diff
@@ -649,6 +649,7 @@ When this method is invoked, the user agent MUST execute the following algorithm
         1. Otherwise, [=list/Append=] |C| to |excludeCredentialDescriptorList|.
     1. [=In parallel=], invoke the [=authenticatorMakeCredential=] operation on |authenticator| with |rpId|,
         |clientDataHash|, |options|.{{MakeCredentialOptions/rp}}, |options|.{{MakeCredentialOptions/user}},
+        <code>|options|.|authenticatorSelection|.{{AuthenticatorSelectionCriteria/requireResidentKey}}</code>,
```
Comment (jyasskin) [inline index.bs]: I think you can replace `|authenticatorSelection|` with `{{MakeCredentialOptions/authenticatorSelection}}` in order to remove it from `Ignored Vars`, but IM me if that gives you bikeshed errors.

**Comment by selfissued** [comment]
This is now addressed by PR #460 .

File: index.bs [hunk 3]
```diff
@@ -643,12 +649,13 @@ When this method is invoked, the user agent MUST execute the following algorithm
...
     1. Let |excludeCredentialDescriptorList| be a new [=list=].
-    1. [=list/For each=] credential descriptor |C| in <code>|options|.{{MakeCredentialOptions/excludeCredentials}}</code>:
+    1. [=list/For each=] credential descriptor |C| in <code>|options|.{{MakePublicKeyCredentialOptions/excludeCredentials}}</code>:
         1. If <code>|C|.{{transports}}</code> [=list/is not empty=], and |authenticator| is connected over a transport not
             mentioned in <code>|C|.{{transports}}</code>, the client MAY [=continue=].
         1. Otherwise, [=list/Append=] |C| to |excludeCredentialDescriptorList|.
     1. [=In parallel=], invoke the [=authenticatorMakeCredential=] operation on |authenticator| with |rpId|,
-        |clientDataHash|, |options|.{{MakeCredentialOptions/rp}}, |options|.{{MakeCredentialOptions/user}},
+        |clientDataHash|, |options|.{{MakePublicKeyCredentialOptions/rp}}, |options|.{{MakePublicKeyCredentialOptions/user}},
+        <code>|options|.{{MakePublicKeyCredentialOptions/authenticatorSelection}}.{{AuthenticatorSelectionCriteria/requireResidentKey}}</code>,
```
Comment (equalsJeffH) [inline index.bs]: the param ordering here does not match the ordering down in [Line 1556](https://github.com/w3c/webauthn/pull/502/files#diff-ec9cfa5f3f35ec1f84feb2e59686c34dR1556) thru 1559.  I suggest they match ordering-wise. thx.

**Comment by equalsJeffH** [comment]
Modulo my line-level comment above, this looks ok I think (am sorta rushed right now and the conflict-fixing added in a bunch of noise into the diff...)

**Comment by selfissued** [comment]
Jeff. maybe I'm tired, but I couldn't figure out the ordering mismatch.  Can you submit a PR fixing this and I'll quickly review it and approve it?  Thanks.

**Pull request merged**